app/controllers/cases/offender_sar_controller.rb

```diff
@@ -330,11 +330,21 @@ def build_case_from_session(correspondence_type)
       # similar workaround needed for request dated
       request_dated_exists = values.fetch("request_dated", false)
       values["request_dated"] = nil unless request_dated_exists
-      values["current_state"] = "rejected" if params.key?("rejected")
+
+      rejected_set_current_state(params, values)
 
       correspondence_type.new(values).decorate
     end
 
+    def rejected_set_current_state(params, values)
+      case params["rejected"]
+      when "true"
+        values["current_state"] = "rejected"
+      when "false"
+        values["current_state"] = "data_to_be_requested"
```

Comment:
Not sure whether its good to hard code the 'data_to_be_requested" state, however not sure how else to do it

Reply:
What would happen if you removed the value?
```
values.delete("current_state")
```